_posts/2025-12-01-idempotency-in-apis.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title: Expanding idempotency superpower to application interfaces (like REST APIs)
+image: /assets/img/cloth_submerged_in_water_by_harbor_with_bus.webp
+image_alt: "Midjourney prompt: A hand is submerging a small beige cloth into water. In the background drives a bus by the harbor promenade. Simple picture. Teal is the primary color. - Did a few edits to remove 'noise'."
+description: "With idempotency, fault-tolerant REST APIs are possible, even though retry strategies needed to mitigate outages, expose the 'exactly once delivery' problem."
+categories:
+- Programming
+tags:
+- Clojure
+- idempotency
+- exactly-once
+- API
+comments: true
+support: true
+excerpt_separator: <!--more-->
+---
+
+Applying retry strategies on API calls to make them more robust
+often brings its own set of problems.
+Instead of occasionally losing a call creating an entity,
```

Comment:
Occasionally, rather than losing an entity-creating call, you end up with two of them.